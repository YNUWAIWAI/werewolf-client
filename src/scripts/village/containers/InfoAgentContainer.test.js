// @flow
import InfoAgentContainer from './InfoAgentContainer'
import IntlProviderContainer from './IntlProviderContainer'
import {Provider} from 'react-redux'
import React from 'react'
import fakeStore from './fakeStore'
import {mount} from 'enzyme'

describe('<InfoAgentContainer />', () => {
  test('initialState', () => {
    const store = fakeStore()
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <InfoAgentContainer />
        </IntlProviderContainer>
      </Provider>
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
  test('agents.mine exists', () => {
    const store = fakeStore(
      {
        agents: {
          all: [],
          mine: {
            '@context': 'https://werewolf.world/context/0.2/agent.jsonld',
            '@id': 'https://licos.online/state/0.2/village#3/agent#1',
            'id': 1,
            'image': 'https://werewolf.world/image/0.2/Walter.jpg',
            'isAChoice': false,
            'isMine': true,
            'name': {
              'en': 'Walter',
              'ja': 'ヴァルター'
            },
            'status': 'alive'
          }
        },
        language: 'en'
      }
    )
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <InfoAgentContainer />
        </IntlProviderContainer>
      </Provider>
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
})
