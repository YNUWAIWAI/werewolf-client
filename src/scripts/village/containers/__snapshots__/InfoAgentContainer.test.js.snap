// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoAgentContainer /> agents.mine exists 1`] = `"<div class=\\"info--agent \\"><img class=\\"info--agent--image\\" src=\\"https://werewolf.world/image/0.2/Walter.jpg\\"><span class=\\"info--agent--name\\">Walter</span></div>"`;

exports[`<InfoAgentContainer /> initialState 1`] = `"<div class=\\"info--agent \\"><img class=\\"info--agent--image\\" src=\\"\\"><span class=\\"info--agent--name\\"></span></div>"`;
